lib/two_factor_authentication/models/two_factor_authenticatable.rb

```diff
@@ -73,8 +74,14 @@ def totp_enabled?
           respond_to?(:otp_secret_key) && !otp_secret_key.nil?
         end
 
+        def confirm_totp_secret(secret, code, options = {})
+          return false unless authenticate_totp(code, {otp_secret_key: secret})
+          self.otp_secret_key = secret
+          true
+        end
+
         def generate_totp_secret
-          self.otp_secret_key = ROTP::Base32.random_base32
+          ROTP::Base32.random_base32
         end
```

Comment:
This method is not called anywhere here, which implies that it's meant to be called from the implementation app. If so, please update the README to explain how this new feature is meant to be used. I'm assuming that the purpose of this change is to differentiate between users who have tried to enable TOTP but failed from those who have tried and succeeded. If so, I'm not convinced that this is the best way to do this. I'll have to think about this some more to see if there's a better way. Perhaps by adding a column called `unconfirmed_otp_secret_key` and then copying it over to `otp_secret_key` once it's confirmed?